src/vm/core_test.zig

```diff
@@ -2027,3 +2027,127 @@ test "CairoVM: markAddressRangeAsAccessed should return an error if the run is n
         vm.markAddressRangeAsAccessed(Relocatable.init(0, 0), 3),
     );
 }
+
+test "CairoVM: opcodeAssertions should throw UnconstrainedAssertEq error" {
+    var instruction = testInstruction;
+    instruction.opcode = .AssertEq;
+
+    const operands = OperandsResult{
+        .dst = .{ .felt = Felt252.fromInteger(8) },
+        .res = null,
+        .op_0 = .{ .felt = Felt252.fromInteger(9) },
+        .op_1 = .{ .felt = Felt252.fromInteger(10) },
+        .dst_addr = .{},
+        .op_0_addr = .{},
+        .op_1_addr = .{},
+        .deduced_operands = 0,
+    };
+
+    var vm = try CairoVM.init(std.testing.allocator, .{});
+    defer vm.deinit();
+
+    try expectError(
+        CairoVMError.UnconstrainedResAssertEq,
+        vm.opcodeAssertions(&instruction, operands),
+    );
+}
+
+test "CairoVM: opcodeAssertions instructions failed - should throw DiffAssertValues error" {
+    var instruction = testInstruction;
+    instruction.opcode = .AssertEq;
+
+    const operands = OperandsResult{
+        .dst = .{ .felt = Felt252.fromInteger(9) },
```

Comment:
Here you can use directly: `.dst = MaybeRelocatable.fromU64(9)`, same for the other ones.